pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStreamingDispatcherMultipleConsumers.java

```diff
@@ -113,12 +113,14 @@ public synchronized void readEntryComplete(Entry entry, PendingReadEntryRequest
             // in a separate thread, and we want to prevent more reads
             sendInProgress = true;
             dispatchMessagesThread.execute(safeRun(() -> {
-                if (sendMessagesToConsumers(readType, Lists.newArrayList(entry), ctx.isLast())) {
+                if (sendMessagesToConsumers(readType, Lists.newArrayList(entry),
+                        readType == ReadType.Normal || ctx.isLast())) {
```

Comment:
Only normal-type reads will run here, so `readType == ReadType.Normal` will always be true.